src/lib/core/components/computations/Utils.ts

```diff
@@ -47,3 +50,177 @@ function createRandomString(numChars: number) {
     .toString(36)
     .slice(2, numChars + 2);
 }
+
+export function assertComputationWithConfig<ConfigType>(
+  computation: Computation,
+  decoder: t.Type<Computation, unknown, unknown>
+): asserts computation is Computation<ConfigType> {
+  const onLeft = (errors: t.Errors) => {
+    throw new Error(`Invalid computation configuration: ${errors}`);
+  };
+  const onRight = () => null;
+  pipe(decoder.decode(computation), fold(onLeft, onRight));
+}
+
+export function useConfigChangeHandler<ConfigType>(
+  analysisState: AnalysisState,
+  computation: Computation<ConfigType>,
+  visualizationId: string
+) {
+  const { url } = useRouteMatch();
+  const history = useHistory();
+  return useCallback(
+    (
+      propertyName: keyof ConfigType,
+      value: ConfigType[typeof propertyName]
+    ) => {
+      const { configuration } = computation.descriptor;
+      handleConfigurationChanges(
+        analysisState,
+        computation,
+        { ...configuration, [propertyName]: value },
+        visualizationId,
+        url,
+        history
+      );
+    },
+    [analysisState, computation, history, url, visualizationId]
+  );
+}
+
+function handleConfigurationChanges<ConfigType>(
+  analysisState: AnalysisState,
+  computation: Computation,
+  updatedConfiguration: ConfigType,
+  visualizationId: string,
+  url: string,
+  history: RouterChildContext['router']['history']
+) {
+  // when a config value changes:
+  // 1. remove viz from current computation
+  // 2. check if the newConfig exists
+  // Y? move viz to the found computation, "existingComputation"
+  // N? create new computation
```

Comment:
Do we need 3. check if the original computation is now empty and delete if it is? From a user's perspective, all is working great, so perhaps 3. is unnecessary.